### ISSUE TITLE
ABGR8888 causes DisplayLink reconnection loop on Wayland sessions

### ISSUE THREAD

**Issue by YourSandwich** [DisplayLink/evdi#459]
# Freeze Issue with DisplayLink Monitors on Wayland with ARGB8888 Based Compositors

## Summary
A significant issue has been identified affecting users of the Plasma 6 environment when connecting monitors through a Lenovo dud9011d1 docking station via DisplayLink to a Lenovo ThinkBook 14 G3 laptop. The core of the problem lies in the freezing of `kwin_wayland` when the compositor uses ARGB8888 pixel format. This freezing manifests shortly after the monitors are connected, either becoming static or entering a loop of continuous reinitialization.

- **Laptop Model:** Lenovo ThinkBook 14 G3, AMD GPU
- **Docking Station:** Lenovo dud9011d1, connected over USB-C with DisplayLink protocol
- **Monitors:** Two 1920x1080, 75hz, Scaling 100%
- **Kernel Drivers on X11:** `evdi` (version 1.14.2) and `displaylink`

## Steps to Reproduce
1. Initiate a Plasma Wayland session (applicable to both Plasma 5 and 6).
2. Connect monitors using the Lenovo dud9011d1 docking station.
3. Attach the docking station to the laptop via DisplayLink.

## Observed Behavior
- Monitors may momentarily function (1-2 seconds) before the display freezes.
- While the laptop's main screen remains responsive, the session is prone to frequent freezes, or the external monitors may repeatedly reinitialize.

## System Information
- **Kernel Versions Tested:** 6.7.6 and 6.1.59
- **Operating System:** ArchLinux (as of 27.02.2024)
- **KDE Plasma Version:** 5.27.8, with the issue persisting into Plasma 6.
- **KDE Frameworks Version:** 5.110.0
- **Qt Version:** 5.15.11

## Resolution
The issue was effectively mitigated by switching the pixel format from ARGB8888 to XRGB8888, utilizing the `KWIN_DRM_NO_AMS=1` environment variable. This adjustment resulted in the stable operation of all connected displays without the prior freezing or reinitialization issues.

## Additional Context
Through extensive discussions and diagnostics tracked in the KDE bug report ([#475922](https://bugs.kde.org/show_bug.cgi?id=475922)), the problem is suggested to stem from a driver-level inconsistency specifically related to the handling of the ARGB8888 pixel format by the DisplayLink technology. Switching to the XRGB8888 format, which Xorg usually uses by default, has been a solid quick fix for Plasma 6.

**Comment by YourSandwich** [comment]
More details available at the KDE Issue Tracker -> https://bugs.kde.org/show_bug.cgi?id=475922

**Comment by stollr** [comment]
I have the same issue on my [**TUXEDO Aura 15 Gen1 Notebook**](https://www.tuxedocomputers.com/de/Linux-Hardware/Linux-Notebooks/15-16-Zoll/TUXEDO-Aura-15-Gen1.tuxedo):
CPU: AMD Ryzen 7 4700U
Graphic: AMD Radeon™ RX Vega Graphics

Docking station:
[**i-tec USB 3.0 / USB-C / Thunderbolt 3, 3x 4K Docking Station + Power Delivery 85W**](https://i-tec.pro/de/produkt/catriple4kdockpd-3/) (CATRIPLE4KDOCKPD):

OS: KDE neon 6.0 (based on Ubuntu 22.04)
Kernel: Linux version 6.5.0-10022-tuxedo

**Comment by YourSandwich** [comment]
You can fix that temporary for Plasma by setting `KWIN_DRM_NO_AMS=1` env Variable
This will enforce XRGB8888 pixel formatting.

**Comment by Contik** [comment]
Also affected, workaround works:
```
# grep -s . /sys/devices/evdi/*
/sys/devices/evdi/count:2
/sys/devices/evdi/loglevel:4
/sys/devices/evdi/version:1.14.2
```
- HP USB-C/A Universal Dock G2, model L61608-001
- HP ProBook 455 G10, model 854L2ES#ABD, has a Ryzen 7 7730U Barceló-R family CPU and an integrated Vega CPU with Graphics Core Next (GCN) 5 instruction set

**Comment by stollr** [comment]
@YourSandwich Thank you! Your suggested workaround works for me :-)

**Comment by Sino-Huang** [comment]
Sorry to trouble  but I do not understand the effect of` KWIN_DRM_NO_AMS=1`
Will it affect any other things in the computer?

**Comment by Zamundaaa** [comment]
Right now it reduces the maximum bits per color on all displays to 8, and may increase latency a tiny bit. As a temporary workaround it's fine to use

**Comment by Sino-Huang** [comment]
Greeting, does Plasma 6.0.3 solve the problem?

**Comment by YourSandwich** [comment]
No

**Comment by adam-kaleta** [comment]
Hello 
Looks like that the latest released evdi=1.14.4-80 (released on 11.04.2024) is fixing this problem.
Could you please try to upgrade evdi and recheck if it is fixing this problem on your side as well?

**Comment by displaylink-emajewsk** [comment]
Closing. Please reopen or create new issue if the problem reappears
